tests/e2e-playwright/tests/jupyterlabs/jupyterlab.py

```diff
@@ -0,0 +1,142 @@
+# pylint: disable=redefined-outer-name
+# pylint: disable=unused-argument
+# pylint: disable=unused-variable
+# pylint: disable=too-many-arguments
+# pylint: disable=too-many-statements
+# pylint: disable=unnecessary-lambda
+
+import json
+import logging
+import re
+from collections.abc import Callable
+from dataclasses import dataclass
+from typing import Final, Literal
+
+from playwright.sync_api import Page, WebSocket
+from pydantic import ByteSize
+from pytest_simcore.logging_utils import log_context
+from pytest_simcore.playwright_utils import MINUTE, SECOND, ServiceType
+
+_WAITING_FOR_SERVICE_TO_START: Final[int] = (
+    10 * MINUTE
+)  # NOTE: smash is 13Gib, math 2Gib
+_WAITING_TIME_FILE_CREATION_PER_GB_IN_TERMINAL: Final[int] = 10 * SECOND
+_SERVICE_NAME_EXPECTED_RESPONSE_TO_WAIT_FOR: Final[dict[str, re.Pattern]] = {
+    "jupyter-math": re.compile(r"/api/contents/workspace"),
+    "jupyter-smash": re.compile(r"/api/contents/workspace"),
+    "jupyter-smash-8-0-0": re.compile(r"/api/contents/workspace"),
+    "jupyter-octave-python-math": re.compile(r"/api/contents"),
+}
+
+_SERVICE_NAME_TAB_TO_WAIT_FOR: Final[dict[str, str]] = {
+    "jupyter-math": "README.ipynb",
+    "jupyter-smash": "README.ipynb",
+    "jupyter-smash-8-0-0": "README.ipynb",
+}
+
+
+@dataclass
+class _JLabTerminalWebSocketWaiter:
+    expected_message_type: Literal["stdout", "stdin"]
+    expected_message_contents: str
+
+    def __call__(self, message: str) -> bool:
+        with log_context(logging.DEBUG, msg=f"handling websocket {message=}"):
+            decoded_message = json.loads(message)
+            if (
+                self.expected_message_type == decoded_message[0]
+                and self.expected_message_contents in decoded_message[1]
+            ):
+                return True
+
+            return False
+
+
+@dataclass
+class _JLabWaitForTerminalWebSocket:
+    def __call__(self, new_websocket: WebSocket) -> bool:
+        with log_context(logging.DEBUG, msg=f"received {new_websocket=}"):
+            if "terminals/websocket" in new_websocket.url:
+                return True
+
+            return False
+
+
+def test_jupyterlab(
```

Comment:
If we are going to have one test per test module, perhaps it should just be called `test`

Reply:
well since pytest auto-discovers, I guess it does not hurt to have that like this. Also the assumption that there is only 1 test per file, I don't know if that holds.